psbt/src/csval/tapret.rs

```diff
@@ -109,53 +119,43 @@ pub enum TapretKeyError {
     /// PSBT_OUT_TAPRET_HOST flag.
     TapretProhibited,
 
-    /// the provided tapret proof is invalid: it has a script longer than 64KiB.
-    InvalidProof,
-
     /// the provided output is not a taproot output and can't host a tapret
     /// commitment.
     NotTaprootOutput,
-}
 
-pub trait OutputTapret {
-    fn is_tapret_host(&self) -> bool;
-    fn set_tapret_host(&mut self) -> Result<(), TapretKeyError>;
-    fn has_tapret_commitment(&self) -> bool;
-    fn tapret_commitment(&self) -> Option<mpc::Commitment>;
-    fn set_tapret_commitment(
-        &mut self,
-        commitment: mpc::Commitment,
-        proof: &TapretPathProof,
-    ) -> Result<(), TapretKeyError>;
-    fn has_tapret_proof(&self) -> bool;
-    fn tapret_proof(&self) -> Option<TapretPathProof>;
+    /// the output contains no valid tapret commitment.
+    NoCommitment,
+
+    /// the value of tapret commitment has invalid length.
+    InvalidCommitment,
+
+    /// Using non-empty taptree is not supported in RGB v0.10. Please update.
```

Comment:
Does this prerequisite remain in later versions?

```suggestion
    /// Using non-empty taptree is not supported in RGB v0.10 or later. Please update.
```

Reply:
Maybe "is not yet supported" without a specific version? At the end this is not even an RGB library...

Reply:
Maybe, we can move this validation to rgb libs?

Reply:
Yes. This should be done right after v0.11 release. For now I just wanted to save time and do this non-trivial thing later when we have at least basic functionality working.

Reply:
> Yes. This should be done right after v0.11 release. For now I just wanted to save time and do this non-trivial thing later when we have at least basic functionality working.

Yes, I agree with you, better focus on this topic later, because the rgb v0.11 topic is the epic of epics =)